xlators/protocol/server/src/server-rpc-fops_v2.c

```diff
@@ -3186,19 +3215,49 @@ int
 server4_lookup_resume(call_frame_t *frame, xlator_t *bound_xl)
 {
     server_state_t *state = NULL;
+    dict_t *xdata = NULL;
 
     state = CALL_STATE(frame);
 
     if (state->resolve.op_ret != 0)
         goto err;
 
-    if (!state->loc.inode)
+    xdata = state->xdata ? dict_ref(state->xdata) : dict_new();
+    if (!xdata) {
+        state->resolve.op_ret = -1;
+        state->resolve.op_errno = ENOMEM;
+        goto err;
+    }
+    if (!state->loc.inode) {
         state->loc.inode = server_inode_new(state->itable, state->loc.gfid);
-    else
+        int ret = dict_set_int32(xdata, GF_NAMESPACE_KEY, 1);
+        if (ret) {
+            gf_msg(THIS->name, GF_LOG_ERROR, ENOMEM, 0,
+                   "dict set (namespace) failed (path: %s), continuing",
+                   state->loc.path);
+            state->resolve.op_ret = -1;
+            state->resolve.op_errno = ENOMEM;
+            goto err;
```

Comment:
xdata will leak here I think, add if (xdata) dict_unref() in err

Reply:
done